src/box/lua/tuple.lua

```diff
@@ -368,6 +403,14 @@ internal.tuple.encode = tuple_encode
 
 -- Public API, additional to implemented in C.
 
+-- new() needs a wrapper in Lua, because format normalization needs to be done
+-- in Lua.
+box.tuple.new = new_tuple
```

Comment:
Is it a problem similar to 75f470cfdb6e0777e2bae671dae382c7e0867728? We just need the `index_base` argument in some costructors, right?

Reply:
No, it has nothing to do with it. Please look at `new_tuple`, `new_format` and `box.internal.space.normalize_format`.

Reply:
I'm about the 'needs to be done in Lua' statement.

I see now, it is not only about one-to-zero index base change (and it is performed for foreign keys, not for other parts of the format). It is also regarding the old `{<type>, <name>}` field_def shape, collation name -> collation id, something about constraints and foreign keys. Ooh, a lot of logic.

So, back to the `box.tuple.new` wrapper in Lua -- the main reason is the normalization of the tuple format definition. I see it now. And it is actually what your comment states.

Maybe, well, not really 'needs to be done', but 'implemented in Lua'. But it goes too picky :)

Closing the discussion. Thanks for the explanation!